packages/perseus-score/src/widgets/widget-registry.ts

```diff
@@ -0,0 +1,108 @@
+import scoreMockWidget from "./mock-widget/score-mock-widget";
+import scoreNumberLine from "./number-line/score-number-line";
+import validateNumberLine from "./number-line/validate-number-line";
+import scoreNumericInput from "./numeric-input/score-numeric-input";
+import scoreOrderer from "./orderer/score-orderer";
+import validateOrderer from "./orderer/validate-orderer";
+import scorePlotter from "./plotter/score-plotter";
+import validatePlotter from "./plotter/validate-plotter";
+import scoreRadio from "./radio/score-radio";
+import validateRadio from "./radio/validate-radio";
+import scoreSorter from "./sorter/score-sorter";
+import validateSorter from "./sorter/validate-sorter";
+import scoreTable from "./table/score-table";
+import validateTable from "./table/validate-table";
+
+import type {
+    WidgetScorerFunction,
+    WidgetValidatorFunction,
+} from "../validation.types";
+
+const widgets = {};
+
+function registerWidget(
+    type: string,
+    scorer: WidgetScorerFunction,
+    validator?: WidgetValidatorFunction,
+) {
+    widgets[type] = {
+        scorer,
+        validator,
+    };
+}
+
+export const getWidgetValidator = (
+    name: string,
+): WidgetValidatorFunction | null => {
+    return widgets[name]?.validator ?? null;
+};
+
+export const getWidgetScorer = (name: string): WidgetScorerFunction | null => {
+    return widgets[name]?.scorer ?? null;
+};
+
+registerWidget(
+    "categorizer",
+    scoreCategorizer as any,
+    validateCategorizer as any,
+);
+registerWidget("cs-program", scoreCSProgram as any);
+registerWidget("dropdown", scoreDropdown as any, validateDropdown as any);
+registerWidget("expression", scoreExpression as any, validateExpression as any);
+registerWidget("grapher", scoreGrapher as any);
+registerWidget("group", scoreGroup as any, validateGroup as any);
+registerWidget("iframe", scoreIframe as any);
+registerWidget("input-number", scoreInputNumber as any);
+registerWidget("interactive-graph", scoreInteractiveGraph as any);
+registerWidget(
+    "label-image",
+    scoreLabelImage as any,
+    validateLabelImage as any,
+);
+registerWidget("matcher", scoreMatcher as any);
+registerWidget("matrix", scoreMatrix as any, validateMatrix as any);
+registerWidget("mock-widget", scoreMockWidget as any, scoreMockWidget as any);
+registerWidget(
+    "number-line",
+    scoreNumberLine as any,
+    validateNumberLine as any,
+);
+registerWidget("numeric-input", scoreNumericInput as any);
+registerWidget("orderer", scoreOrderer as any, validateOrderer as any);
+registerWidget("plotter", scorePlotter as any, validatePlotter as any);
+registerWidget("radio", scoreRadio as any, validateRadio as any);
+registerWidget("sorter", scoreSorter as any, validateSorter as any);
+registerWidget("table", scoreTable as any, validateTable as any);
+
+registerWidget("deprecated-standin", () => scoreNoop(1) as any);
+registerWidget("measurer", () => scoreNoop(1) as any);
+
+registerWidget("definition", scoreNoop as any);
```

Comment:
We need to get rid of `scoreNoop` but I didn't feel like now was the right moment.